File: Problem83/Solution.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	p := head
	for p != nil {
		next := p.Next
		for next != nil && next.Val == p.Val {
			next = next.Next
		}
		p.Next = next
		p = p.Next
	}
	return head
}

File: Problem83/Solution.java
public class Solution
{
    public ListNode deleteDuplicates(ListNode head)
    {
        ListNode p = head;
        while (p != null)
        {
            ListNode next = p.next;
            while (next != null && next.val == p.val)
                next = next.next;
            p.next = next;
            p = p.next;
        }
        return head;
    }
}